examples/system/generic/zynqmp_r5/zynqmp_amp_demo/ipi_shmem_demod.c

```diff
@@ -264,7 +262,8 @@ int ipi_shmem_demod()
 	metal_irq_register(ipi_irq, ipi_irq_handler, ipi_io);
 	metal_irq_enable(ipi_irq);
 	/* initialize remote_nkicked */
-	atomic_init(&remote_nkicked, 1);
+	atomic_flag_clear(&remote_nkicked);
```

Comment:
could you use ATOMIC_FLAG_INIT instead?

Reply:
I agree, there is no reason to have two mem barrier operations during initialization. 
`remote_nkicked = ATOMIC_FLAG_INIT;`
Does it look cleaner? Need to test it internally.